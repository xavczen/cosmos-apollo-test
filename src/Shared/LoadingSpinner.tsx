import React from 'react';

export default () => (
  <svg
    width="200px"
    height="200px"
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 100 100"
    preserveAspectRatio="xMidYMid"
    style={{background: "none"}}
  >
    <g transform="translate(50,50)">
      <g transform="scale(0.7)">
        <g transform="translate(-50,-50)">
          <g transform="rotate(8 50 50)">
            <animateTransform
              attributeName="transform"
              type="rotate"
              calcMode="linear"
              values="0 50 50;360 50 50"
              keyTimes="0;1"
              dur="0.75s"
              begin="0s"
              repeatCount="indefinite"
            />
            <path
              d="M50 50L50 0A50 50 0 0 1 100 50Z"
              fillOpacity="0.8"
              fill="#004d73"
            />
          </g>
          <g transform="rotate(96 50 50)">
            <animateTransform
              attributeName="transform"
              type="rotate"
              calcMode="linear"
              values="0 50 50;360 50 50"
              keyTimes="0;1"
              dur="1s"
              begin="0s"
              repeatCount="indefinite"
            />
            <path
              d="M50 50L50 0A50 50 0 0 1 100 50Z"
              transform="rotate(90 50 50)"
              fillOpacity="0.8"
              fill="#00334e"
            />
          </g>
          <g transform="rotate(184 50 50)">
            <animateTransform
              attributeName="transform"
              type="rotate"
              calcMode="linear"
              values="0 50 50;360 50 50"
              keyTimes="0;1"
              dur="1.5s"
              begin="0s"
              repeatCount="indefinite"
            />
            <path
              d="M50 50L50 0A50 50 0 0 1 100 50Z"
              transform="rotate(180 50 50)"
              fillOpacity="0.8"
              fill="#007fdb"
            />
          </g>
          <g transform="rotate(272 50 50)">
            <animateTransform
              attributeName="transform"
              type="rotate"
              calcMode="linear"
              values="0 50 50;360 50 50"
              keyTimes="0;1"
              dur="3s"
              begin="0s"
              repeatCount="indefinite"
            />
            <path
              d="M50 50L50 0A50 50 0 0 1 100 50Z"
              transform="rotate(270 50 50)"
              fillOpacity="0.8"
              fill="#dbebfa"
            />
          </g>
        </g>
      </g>
    </g>
  </svg>
);
