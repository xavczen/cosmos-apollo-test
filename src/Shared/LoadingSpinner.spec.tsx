import React from 'react';
import renderer from 'react-test-renderer';

import LoadingSpinner from './LoadingSpinner';

describe('LoadingSpinner', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LoadingSpinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
