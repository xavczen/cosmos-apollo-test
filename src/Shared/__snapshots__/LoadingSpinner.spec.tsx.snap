// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly 1`] = `
<svg
  height="200px"
  preserveAspectRatio="xMidYMid"
  style={
    Object {
      "background": "none",
    }
  }
  viewBox="0 0 100 100"
  width="200px"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    transform="translate(50,50)"
  >
    <g
      transform="scale(0.7)"
    >
      <g
        transform="translate(-50,-50)"
      >
        <g
          transform="rotate(8 50 50)"
        >
          <animateTransform
            attributeName="transform"
            begin="0s"
            calcMode="linear"
            dur="0.75s"
            keyTimes="0;1"
            repeatCount="indefinite"
            type="rotate"
            values="0 50 50;360 50 50"
          />
          <path
            d="M50 50L50 0A50 50 0 0 1 100 50Z"
            fill="#004d73"
            fillOpacity="0.8"
          />
        </g>
        <g
          transform="rotate(96 50 50)"
        >
          <animateTransform
            attributeName="transform"
            begin="0s"
            calcMode="linear"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            type="rotate"
            values="0 50 50;360 50 50"
          />
          <path
            d="M50 50L50 0A50 50 0 0 1 100 50Z"
            fill="#00334e"
            fillOpacity="0.8"
            transform="rotate(90 50 50)"
          />
        </g>
        <g
          transform="rotate(184 50 50)"
        >
          <animateTransform
            attributeName="transform"
            begin="0s"
            calcMode="linear"
            dur="1.5s"
            keyTimes="0;1"
            repeatCount="indefinite"
            type="rotate"
            values="0 50 50;360 50 50"
          />
          <path
            d="M50 50L50 0A50 50 0 0 1 100 50Z"
            fill="#007fdb"
            fillOpacity="0.8"
            transform="rotate(180 50 50)"
          />
        </g>
        <g
          transform="rotate(272 50 50)"
        >
          <animateTransform
            attributeName="transform"
            begin="0s"
            calcMode="linear"
            dur="3s"
            keyTimes="0;1"
            repeatCount="indefinite"
            type="rotate"
            values="0 50 50;360 50 50"
          />
          <path
            d="M50 50L50 0A50 50 0 0 1 100 50Z"
            fill="#dbebfa"
            fillOpacity="0.8"
            transform="rotate(270 50 50)"
          />
        </g>
      </g>
    </g>
  </g>
</svg>
`;
